### ISSUE TITLE
Compatibility issue with openai-node v4.11.0 due to signature change in completions class

### ISSUE THREAD

**Issue by peterdresslar** [vercel/ai#611]
### Description

Summary:
After updating to OpenAI Node library version 4.11.0 (released today), our downstream callers using `ai` streaming are experiencing compatibility issues, particularly with the `OpenAIStream` function. It appears the major cause is a change in the signature of the `Completions` class in the upstream OpenAI SDK, especially concerning the `ChatCompletionChunk` type which is not directly exposed for external use. The attached code example worked fine with version 4.10.0. Now typescript will reject and the code will run in dev but not build or deploy.

---

After updating to version 4.11.0, the same code throws a TypeScript error since the `OpenAIStream` function expects an `AsyncIterable<ChatCompletionChunk>` or `AsyncIterable<Completion>` type. The newly updated signature of `Completions.create()` in openai 4.11.0 returns a `Stream<ChatCompletionChunk>`, which-- although `Stream` implements AsyncIterable in the OpenAI library--is not directly usable here. This, despite the union type for `AsyncIterableOpenAIStreamReturnTypes`:

```typescript
// openai-stream.ts
type AsyncIterableOpenAIStreamReturnTypes = AsyncIterable<ChatCompletionChunk> | AsyncIterable<Completion>;
declare function OpenAIStream(res: Response | AsyncIterableOpenAIStreamReturnTypes, callbacks?: OpenAIStreamCallbacks): ReadableStream;
```

As a result, it seems like types just need to be updated or regenerated to solve this problem--but, it also seems like there could be other breaking changes from the other pretty significant changes in the upstream `Completions` class.

Since `ChatCompletionChunk` is not exposed by either library, the easy and relatively safe seeming type casting workaround is not available.

```typescript
const stream = OpenAIStream(response as AsyncIterable<ChatCompletionChunk>, { //seems like it would work but we don't have access to CCC
    ...
  });
```

  

### Code example

```typescript
const response = await openai.chat.completions.create({
  model: "gpt-3.5-turbo",
  stream: true,
  messages,
});
const stream = OpenAIStream(response); // response is no longer an accepted type
```

### Additional context

The workaround right now is to just fall back to `openai` 4.10.0. However, this seems like a pretty urgent issue since it will break the flow of anybody new to the SDK and following the examples or tutorials. 

This is the same issue as #610, but a bit more broad.

**Comment by MaxLeiter** [comment]
Thanks for the write up. This should be resolved in 2.2.14

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete compatibility failure in existing `OpenAIStream` behavior caused by an upstream type-signature change, not a request for a new capability.

### Evidence
- After upgrading openai-node from 4.10.0 to 4.11.0, TypeScript rejects the value returned by `openai.chat.completions.create()` when passed to `OpenAIStream`, preventing builds and deployments.
- The same code reportedly worked with openai-node 4.10.0, indicating a compatibility regression.
- A maintainer stated that the issue should be resolved in AI SDK 2.2.14.
